internal/manifests/collector/service_test.go

```diff
@@ -321,6 +321,161 @@ func TestMonitoringService(t *testing.T) {
 	})
 }
 
+func TestExtensionService(t *testing.T) {
+	t.Run("when the extension has http endpoint", func(t *testing.T) {
+		params := manifests.Params{
+			Config: config.Config{},
+			Log:    logger,
+			OtelCol: v1beta1.OpenTelemetryCollector{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test",
+				},
+				Spec: v1beta1.OpenTelemetryCollectorSpec{
+					Config: v1beta1.Config{
+						Service: v1beta1.Service{
+							Extensions: []string{"jaeger_query"},
+						},
+						Extensions: &v1beta1.AnyConfig{
+							Object: map[string]interface{}{
+								"jaeger_query": map[string]interface{}{
+									"http": map[string]interface{}{
+										"endpoint": "0.0.0.0:16686",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		actual, err := ExtensionService(params)
+		assert.NotNil(t, actual)
+		assert.NoError(t, err)
+	})
+
+	t.Run("when the extension has grpc endpoint", func(t *testing.T) {
+		params := manifests.Params{
+			Config: config.Config{},
+			Log:    logger,
+			OtelCol: v1beta1.OpenTelemetryCollector{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test",
+				},
+				Spec: v1beta1.OpenTelemetryCollectorSpec{
+					Config: v1beta1.Config{
+						Service: v1beta1.Service{
+							Extensions: []string{"jaeger_query"},
+						},
+						Extensions: &v1beta1.AnyConfig{
+							Object: map[string]interface{}{
+								"jaeger_query": map[string]interface{}{
+									"grpc": map[string]interface{}{
+										"endpoint": "0.0.0.0:16686",
```

Comment:
why this port? The default ports are
```
Query server started	{"kind": "extension", "name": "jaeger_query", "http_addr": "[::]:16686", "grpc_addr": "[::]:16685"}
```

Reply:
It's just used to test whether the function is working correctly or not.